Tests/BugsnagConfigurationTests.m

```diff
@@ -583,6 +574,113 @@ - (void)testMaxBreadcrumb {
+    XCTAssertTrue(config.autoTrackSessions);
+}
+
+- (void)testDefaultBundleVersion {
+    BugsnagConfiguration *config = [[BugsnagConfiguration alloc] initWithApiKey:DUMMY_APIKEY_32CHAR_1];
+    XCTAssertEqualObjects(NSBundle.mainBundle.infoDictionary[@"CFBundleVersion"], config.bundleVersion);
+}
+
+- (void)testDefaultContext {
+    BugsnagConfiguration *config = [[BugsnagConfiguration alloc] initWithApiKey:DUMMY_APIKEY_32CHAR_1];
+    XCTAssertNil(config.context);
+}
+
+- (void)testDefaultEnabledBreadcrumbTypes {
+    BugsnagConfiguration *config = [[BugsnagConfiguration alloc] initWithApiKey:DUMMY_APIKEY_32CHAR_1];
+    XCTAssertEqual(BSGEnabledBreadcrumbTypeAll, config.enabledBreadcrumbTypes);
+}
+
+- (void)testDefaultEnabledErrorTypes {
+    BugsnagConfiguration *config = [[BugsnagConfiguration alloc] initWithApiKey:DUMMY_APIKEY_32CHAR_1];
+
+    XCTAssertTrue(config.enabledErrorTypes.cppExceptions);
+    XCTAssertTrue(config.enabledErrorTypes.machExceptions);
+    XCTAssertTrue(config.enabledErrorTypes.signals);
+    XCTAssertTrue(config.enabledErrorTypes.unhandledExceptions);
+    XCTAssertTrue(config.enabledErrorTypes.unhandledRejections);
+
+    XCTAssertFalse(config.enabledErrorTypes.ooms);
+}
+
+- (void)testDefaultEnabledReleaseStages {
+    BugsnagConfiguration *config = [[BugsnagConfiguration alloc] initWithApiKey:DUMMY_APIKEY_32CHAR_1];
+
+    XCTAssertNil(config.enabledReleaseStages);
+}
+
+- (void)testDefaultEndpoints {
+    BugsnagConfiguration *config = [[BugsnagConfiguration alloc] initWithApiKey:DUMMY_APIKEY_32CHAR_1];
+
+    XCTAssertEqualObjects(@"https://notify.bugsnag.com", config.endpoints.notify);
+    XCTAssertEqualObjects(@"https://sessions.bugsnag.com", config.endpoints.sessions);
+}
+
+- (void)testDefaultMaxBreadcrumbs {
+    BugsnagConfiguration *config = [[BugsnagConfiguration alloc] initWithApiKey:DUMMY_APIKEY_32CHAR_1];
+
+    XCTAssertEqual(25, config.maxBreadcrumbs);
+}
+
+- (void)testDefaultpersistUser {
+    BugsnagConfiguration *config = [[BugsnagConfiguration alloc] initWithApiKey:DUMMY_APIKEY_32CHAR_1];
+
+    XCTAssertTrue(config.persistUser);
+}
+
+- (void)testDefaultRedactedKeys {
+    BugsnagConfiguration *config = [[BugsnagConfiguration alloc] initWithApiKey:DUMMY_APIKEY_32CHAR_1];
+
+    XCTAssertEqual(1, [config.redactedKeys count]);
+    XCTAssertEqualObjects(@"password", [config.redactedKeys allObjects][0]);
+}
+
+- (void)testDefaultReleaseStage {
+    BugsnagConfiguration *config = [[BugsnagConfiguration alloc] initWithApiKey:DUMMY_APIKEY_32CHAR_1];
+#if DEBUG
+    XCTAssertEqualObjects(@"development", config.releaseStage);
+#else
+    XCTAssertEqualObjects(@"production", config.releaseStage);
+#endif
+}
+
+- (void)testDefaultSendThreads {
+    BugsnagConfiguration *config = [[BugsnagConfiguration alloc] initWithApiKey:DUMMY_APIKEY_32CHAR_1];
+
+    XCTAssertEqual(BSGThreadSendPolicyAlways, config.sendThreads);
+}
```

Comment:
I think most - if not all - of these can be lumped into a single test.  There shouldn't be any interaction between values and modifying them should be side-effect free.